### DISCUSSION TITLE
v4.11.1 - Added the Ability to Throw Fault Exception in Flow Actions

### DISCUSSION THREAD

**Discussion by jongpie** [jongpie/NebulaLogger#531] [Announcements]
A huge thank you to @JMercie for working on this release, and congratulations on their first-ever open source contribution! (PR #516). And many thanks also to @sidle for suggesting this great enhancement (issue #489).

# Core Unlocked Package Changes

## Added the Ability to Throw Fault Exceptions in Flow Actions
This release closes #489 by providing a new option when logging in Flow - when logging a Flow fault message (shown as `1` in the screenshot below), you can now have the invocable action automatically re-throw the encountered exception (shown as `2` in the screenshot below). This provides a great way to still display an error to the user & rollback any other DML in the current transaction, while still saving your logging data.

- Updated  all 3 Flow invocable logging actions (Apex classes `FlowLogEntry`, `FlowRecordLogEntry`, and `FlowCollectionLogEntry`) to have a new optional `Boolean` property, `shouldThrowFaultMessageException`. When set to `true`, the provided Flow fault message (using the `String` property `faultMessage`) will be used to create & throw an instance of the standard exception class `System.FlowException`
- Updated `FlowLogger` to handle throwing the exception for all 3 Flow invocable logging actions

![image](https://github.com/jongpie/NebulaLogger/assets/1267157/25aaec88-496f-491a-ba39-33f84550dc15)

# Installation Info

## Core Unlocked Package - no namespace

Full Changelog: https://github.com/jongpie/NebulaLogger/compare/v4.11.0...v4.11.1
- SF CLI: `sf package install --wait 20 --security-type AdminsOnly --package 04t5Y000001TsX4QAK`
- SFDX CLI: `sfdx force:package:install --wait 20 --securitytype AdminsOnly --package 04t5Y000001TsX4QAK`
- Sandbox: https://test.salesforce.com/packaging/installPackage.apexp?p0=04t5Y000001TsX4QAK
- Production: https://login.salesforce.com/packaging/installPackage.apexp?p0=04t5Y000001TsX4QAK

<hr /><em>This discussion was created from the release <a href='https://github.com/jongpie/NebulaLogger/releases/tag/v4.11.1'>Added the Ability to Throw Fault Exception in Flow Actions</a>.</em>

**Comment by mspiep01** [comment]
This is amazing, this will work well for callouts I think.

**Comment by eneag-sf** [comment]
awesome, thank you @jongpie!
